Packs/EmailCommunication/Scripts/SummarizeEmailThreads/SummarizeEmailThreads.py

```diff
@@ -0,0 +1,114 @@
+import demistomock as demisto  # noqa: F401
+from CommonServerPython import *  # noqa: F401
+
+
+def fetch_email_threads(incident_id):
+    """Fetch all Email Threads stored in the current incident context
+    Args:
+        incident_id (str): The incident ID
+    Returns:
+        list of dicts. List of email thread entries
+    """
+    # Get current email threads from context if any are present
+    try:
+        incident_context = demisto.executeCommand("getContext", {'id': incident_id})
+        threads = dict_safe_get(incident_context[0], ['Contents', 'context', 'EmailThreads'])
+
+        if not threads:
+            return None
+
+        # Return a list of dicts, even if there is only one thread entry and the context item is a list
+        threads = [threads] if isinstance(threads, dict) else threads
+
+        return threads
+
+    except Exception as e:
+        print(f'Exception in fetch_email_threads: {e}')
+
+
+def format_threads(email_threads):
+    """Process list of email thread context entries and return a table in MD format
+    Args:
+        email_threads (list of dicts): List of email thread entry dictionaries
+    Returns:
+        md (str): Markdown formatted table
+    """
+    try:
+        thread_summary_md = []
+        for thread_entry in email_threads:
+            # Loop through thread entries.  'EmailCommsThreadNumber' is indexed from 0 so we can use it for list indexing
+            # Trim off thread ID code for readability
+            thread_number = thread_entry['EmailCommsThreadNumber']
+            email_original_subject = thread_entry['EmailSubject'].split('<')[-1].split('>')[1].strip()
+            cc_addresses = thread_entry['EmailCC']
+            bcc_addresses = thread_entry['EmailBCC']
+            recipients = thread_entry['EmailTo']
+
+            if 0 <= int(thread_number) < len(thread_summary_md):
```

Comment:
you are using the thread numbers as an index in the list, and append the table_row for the thread number at the end of the list. If I understand correctly, this code can only work if thread numbers are ordered always 0,1,2,3 and none of them will be missed. or else this indexing: `thread_summary_md[int(thread_number)]` will get the wrong item in the list. is that what you meant to do here?

Reply:
Done.

Reply:
New entries in the `EmailThreads` context key should always be created with the `EmailCommsThreadNumber` starting at zero, with each new email thread not related to an existing entry getting its own new number incrementing by one each time.

There is no function within this content pack that would ever remove entries from `EmailThreads`, however I can see how it might be possible for actions taken by other tasks to make changes to this context key.

I have updated this function to use a dictionary of dictionaries to provide reliable indexing, then convert that to a list of dictionaries for MD formatting.